amd_openvx_extensions/amd_rpp/source/tensor/Slice.cpp

```diff
@@ -0,0 +1,306 @@
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "internal_publishKernels.h"
+
+struct SliceLocalData {
+    vxRppHandle *handle;
+    Rpp32u deviceType;
+    RppPtr_t pSrc;
+    RppPtr_t pDst;
+    Rpp32s *pAnchor;
+    Rpp32s *pShape;
+    Rpp32f *pFillValues;
+    Rpp32u policy;
+    Rpp32u *pSrcDims;
+    RpptDescPtr pSrcDesc;
+    RpptDescPtr pDstDesc;
+    RpptGenericDescPtr pSrcGenericDesc;
+    RpptGenericDescPtr pDstGenericDesc;
+    Rpp32u *pSrcRoi3D;
+    RpptRoiType roiType;
+    vxTensorLayout inputLayout;
+    size_t inputTensorDims[RPP_MAX_TENSOR_DIMS];
+    size_t outputTensorDims[RPP_MAX_TENSOR_DIMS];
+};
+
+void updateDestinationRoi(SliceLocalData *data, unsigned *src_roi, unsigned *dst_roi, const vx_reference parameters[]) {
+    // Query the Anchor / Shape Tensor Dims
+    RppSize_t num_dims;
+    size_t anchorTensorDims[RPP_MAX_TENSOR_DIMS];
+    vxQueryTensor((vx_tensor)parameters[4], VX_TENSOR_NUMBER_OF_DIMS, &num_dims, sizeof(num_dims));
+    vxQueryTensor((vx_tensor)parameters[4], VX_TENSOR_DIMS, &anchorTensorDims, sizeof(vx_size) * num_dims);
+    uint total_anchor_dims = 1;
+    for (uint n = 0; n < num_dims; n++)
+        total_anchor_dims *= anchorTensorDims[n];
+
+    total_anchor_dims /= anchorTensorDims[0];
+
+    if (total_anchor_dims == 1) {
+        // if input is 1D - shape will be of size (batchSize) or (batchSize, 1) - so fill only 1st dim of length in dst_roi from buffer and set 2nd dim of length to 1
+        for (unsigned i = 0, j = 0; i < data->inputTensorDims[0]; i++, j += 4) {
+            dst_roi[j] = data->pAnchor[i];
+            dst_roi[j + 1] = 0;
+            dst_roi[j + 2] = data->pShape[i];
+            dst_roi[j + 3] = 1;
+        }
+    } else {
+        // if input is nD - shape will be of size (batchSize * n) - so fill dst_roi using both values
+        unsigned *tensor_shape = dst_roi;
+        uint anchor_index = 0, shape_index = 0;
+        for (unsigned i = 0; i < data->inputTensorDims[0]; i++) {
+            anchor_index = i * total_anchor_dims * 2;
+            shape_index = anchor_index + total_anchor_dims;
+            for (unsigned d = 0; d < total_anchor_dims; d++) {
+                tensor_shape[anchor_index + d] = data->pAnchor[data->inputTensorDims[1] * i + d];
```

Comment:
data->inputTensorDims[1] * i is constant for d loop at L73
Please declare a variable after L72 uint32_t
int idx = [data->inputTensorDims[1] * i; and use this idx in d loop